packages/astro/src/runtime/server/index.ts

```diff
@@ -480,11 +480,11 @@ export async function renderHead(result: SSRResult): Promise<string> {
 			}
 			return renderElement('script', {
 				...script,
-				props: { ...script.props, 'astro-script': result._metadata.pathname + '/script-' + i },
+				props: { ...script.props, 'astro-script': result._metadata.legacyBuild ? result._metadata.pathname + '/script-' + i : undefined },
 			});
 		});
 	if (needsHydrationStyles) {
-		styles.push(renderElement('style', { props: { 'astro-style': true }, children: 'astro-root, astro-fragment { display: contents; }' }));
+		styles.push(renderElement('style', { props: { 'astro-style': result._metadata.legacyBuild ? true : undefined }, children: 'astro-root, astro-fragment { display: contents; }' }));
```

Comment:
Just a cleanup: `astro-style` prop won't be needed once `legacyBuild` is dropped, so this just cuts down on some output noise in the static build.